### DISCUSSION TITLE
Output `compile_commands.json`

### DISCUSSION THREAD

**Discussion by oraqlle** [cabinpkg/cabin#763] [Q&A] [answered]
Does Poac support outputting the `compile_commands.json` file?

**Comment by ken-matsui** [comment, marked as answer]
@oraqlle No

**Reply by oraqlle** [reply]
@ken-matsui Is there something similar that can be used?

**Reply by ken-matsui** [reply]
@oraqlle Poac directly outputs [`ninja.build`](https://github.com/poac-dev/poac/blob/ea7c2fc343787c46c251981378f222f273f288d9/include/poac/core/builder/manifest.hpp#L18) (this should have been `build.ninja` though). This might be similar to `compile_commands.json`.

**Reply by oraqlle** [reply]
I look into it.

**Reply by oraqlle** [reply]
So the `ninja.build` cannot be used as a compilation database but ninja can use it to generate one. I created a very simple script to generate it quickly. In the future it would be awesome if this was automatically generated.

```sh
# makedb.sh

#! /usr/bin/bash

[ -d "poac-out/debug" ] && ninja -C ./poac-out/debug -f ninja.build -t compdb > ./poac-out/debug/compile_commands.json
[ -d "poac-out/release" ] && ninja -C ./poac-out/release -f ninja.build -t compdb > ./poac-out/release/compile_commands.json

```

**Reply by ken-matsui** [reply]
@oraqlle ~~Should we generate the compilation database every time we build? Or do we just need it when we want?~~ I'm going to implement building the db every time.

**Comment by ken-matsui** [comment]
@oraqlle I implemented: https://github.com/poac-dev/poac/pull/767

**Reply by oraqlle** [reply]
Awesome, I'll give it a go.

**Reply by oraqlle** [reply]
Output database works but it appears to use unnecessary slashes (`\`) on Linux for paths. I imagine this is because Windows uses `\` as a path separator which needs to be escaped in the string.
